game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,106 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  enum DiceType {
+    // Use AA dice (attackAA, offensiveAttackAA)
+    AA,
+    // Use bombard dice (bombard or attack if bombard is empty)
+    BOMBARD,
+    // Use normal dice (attack, defense)
+    NORMAL
+  }
+
+  /** The name of this unit ability that will be shown in the Battle UI */
+  @NonNull String name;
+
+  /** The unit types that have this ability */
+  @NonFinal Collection<UnitType> attachedUnitTypes;
+
+  /** The unit types that can be targeted */
+  @NonNull Collection<UnitType> targets;
+
+  /** The type of dice that will be rolled. */
+  @Builder.Default DiceType diceType = DiceType.NORMAL;
+
+  /** The side(s) that the unit needs to be on to use the ability */
+  @Builder.Default
+  Collection<BattleState.Side> sides = List.of(BattleState.Side.OFFENSE, BattleState.Side.DEFENSE);
+
+  /** How many combat rounds this ability can be used. -1 means unlimited */
+  @Builder.Default int round = -1;
```

Comment:
Would 'round == 0' be useful at all?

The magic number of '-1' sticks out (SRP issues). Any thoughts to perhaps make this "Integer.MAX" (are we going to see 32 bit vs 64 bit compatibility issues if this value differs for one player or another). If the value were very large, then the at least the special case logic for '-1' would drop away and it would instead be effectively unlimited.

Reply:
I didn't realize you were retrofitting this to existing values. There are a number of original design decisions that would be great to change, but we are stuck with them.

Reply:
I see this work as unifying how unit abilities work between bombard, aa, firststrike, and general phase.  There are several ways to indicate targets and other abilities and so I'd like to unify it both in the engine and in the xml.  So all the existing properties still need to be handled with these new abilities so that the engine only has to care about the new abilities and not the old properties.

But I could probably change the old `-1` to be INTEGER.MAX when converting the property.

Reply:
Done in https://github.com/triplea-game/triplea/pull/8646/commits/a6ea214333c993baefa8d1a9a5d95038354190f6